zotero_docai_pipeline/clients/pageindex_client.py

```
@@ -0,0 +1,1192 @@
+                    # Still processing: increment attempts and add back to queue
+                    state["attempts"] += 1
+                    state["last_poll_time"] = time.time()
+                    pending_docs.append(doc_id)  # Add to end of queue (FIFO)
+                    logger.warning(
+                        f"Document {doc_id} still processing "
+                        f"(attempt {state['attempts']}/"
+                        f"{self.config.polling_max_attempts}), will retry"
+                    )
+                else:
+                    # Permanent error: log and remove from pending
+                    error_msg = f"Document {doc_id} failed: {str(e)}"
```

Ruff (0.15.0)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)